src/block_cache/block_io.c

```diff
@@ -406,9 +406,10 @@ __wt_blkcache_read_multi(WT_SESSION_IMPL *session, WT_ITEM **buf, size_t *buf_co
     WT_BTREE *btree;
     WT_DECL_ITEM(ctmp);
     WT_DECL_ITEM(etmp);
+    WT_DECL_ITEM(tmp);
```

Comment:
This warrants a comment. The code here is that a little sneaky in that one branch treats `tmp` as a single item and another branch treats it as an array. That continues to work because `WT_DECL_ITEM` declares it as `WT_ITEM *tmp`, but now we're hiding that.

Reply:
The code around this place is a bit messy so I've decided to move this fix to another ticket, please see the last commit.